app/Http/Controllers/ProductController.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Models\Product;
+use App\Traits\Common;
+
+class ProductController extends Controller
+{
+    use Common;
+    /**
+     * Display a listing of the resource.
+     */
+    public function indexR()
+    {
+
+
+        $products = Product::orderBy('created_at', 'desc')->take(3)->get();
+        return view('index', compact('products'));
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     */
+    public function create()
+    {
+        return view('add_products');
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     */
+    public function store(Request $request)
+    {
+        $data = $request->validate([
+            'title' => 'required|string|max:255',
+            'short_description' => 'required|string|max:1000',
+            'price' => 'required|numeric',
+            'image' => 'nullable|image|mimes:jpeg,png,jpg,gif,svg|max:2048',
```

Comment:
why image is `nullable` here, all products must have image

we make it nullable in update as we can update it or not keep the old image, but in create we have to provide an image